### PR TITLE
A 809/publish artifact to maven central

### PR DESCRIPTION
**Pull request by shaileshp0110**

### CODE REVIEW

File: .circleci/publish.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+set -eu
+
+echo $GPG_KEY | base64 --decode | gpg --batch --import --passphrase "$GPG_PASSPHRASE"
+
+gpg --passphrase $GPG_PASSPHRASE --batch --yes -a -b LICENSE
+
+mill -i src.io.iohk.decco.publish \
+--sonatypeCreds "$OSS_USERNAME:$OSS_PASSWORD"  \
+--gpgPassphrase "$GPG_PASSPHRASE" \
```
Comment (AlexITC) [inline .circleci/publish.sh:13]: As this is a public repository, what prevents people from submitting a PR containing code like `println(java.lang.System.getenv("GPG_PASSPHRASE"))` and looking into the logs to see the printed key?

File: .circleci/publish.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+--gpgKeyName "$GPG_KEY_ID" \
+--release false
+
+mill -i src.io.iohk.decco.auto.publish \
```
Comment (AlexITC) [inline .circleci/publish.sh]: Have you tried `mill -i __.publish ...` for publishing all modules in a single call?

Comment (shaileshp0110) [reply .circleci/publish.sh]: That will release other modules like test etc as well I am not sure we need that to be released

File: .circleci/publish.sh (hunk 1 above)
Comment (shaileshp0110) [reply .circleci/publish.sh:13]: will they have access to circle ci

**Comment by krcz** [comment]
This is going to release code in feature branches as well, which we should avoid. There is example solution for that in my branch in scalanet.

File: .circleci/publish.sh (hunk 2 above)
Comment (AlexITC) [reply .circleci/publish.sh]: In fact, it does release the tests while I expect mill not to do it, thanks.

File: .circleci/publish.sh (hunk 1 above)
Comment (AlexITC) [reply .circleci/publish.sh:13]: Marcin has already addressed this concern in his PR.

**Pull request merged**